src/ResponseParser.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace React\HttpClient;
+
+class ResponseParser
+{
+    public function parse($raw)
+    {
+        if (empty($raw) or false === strpos($raw, "\r\n\r\n")) {
+            return false;
+        }
+
+        list($head, $body) = explode("\r\n\r\n", $raw, 2);
+
+        $lines = explode("\r\n", $head);
+        $first_line = array_shift($lines);
+
+        if (!strpos($first_line, ' ')) {
+            return false;
+        }
+
+        list($http, $code, $reason) = explode(' ', $first_line.' ', 3);
+
+        if (!strpos($http, '/') or (int)$code < 100 or (int)$code >= 1000) {
+            return false;
+        }
+
+        list($protocol, $version) = explode('/', $http, 2);
+
+        if ($protocol !== 'HTTP' or $version !== '1.0') {
```

Comment:
Not sure about the pinning down on 1.0. Using it to talk to 1.1 servers a lot and that works fine. Not sure what their responses are but blocking 1.1 responses while they used to work in previous 0.4.x versions could be a BC break. Will investigate for you :).